### PR TITLE
Ezpz cu 44dbec8: Create Password Field

### PR DESCRIPTION
**Pull request by KevinLaCarrubba**
- Created a password field input type.
- Added icon to toggle view/hide password.
- Added PasswordField to register page to test. 
*****
**SideTask**
- Moved /components/form/formfield to /components/form/fields/StringField
This will allow easier creation of specific field types such as PasswordField, NumberField ect.
- Fixed register page mobile css.

### CODE REVIEW

**Comment by arfusop** [comment]
Task linked: [CU-44dbec8 Create PasswordField component](https://app.clickup.com/t/44dbec8)

**Review by SamiFdev — APPROVED** [review]

File: components/form/fields/PasswordField/index.tsx [hunk 1]
```diff
@@ -0,0 +1,66 @@
...
+import { FaEye, FaEyeSlash } from 'react-icons/fa'
+
+type PasswordFieldProps = {
+    field: inputType
+    id: string
+    name: string
+    placeholder: string
+    onChange: ChangeEventHandler
+    required?: boolean
+    minLength?: number
```
Comment (arfusop) [inline components/form/fields/PasswordField/index.tsx:13]: Password field should require a minimum length. We will never have a password that doesnt have a min 😄

**Review by arfusop — CHANGES_REQUESTED** [review]

File: components/form/fields/PasswordField/index.tsx (hunk 1 above)
Comment (KevinLaCarrubba) [reply components/form/fields/PasswordField/index.tsx:13]: i was thinking that and required to take off
and always leave it. makes sense

File: components/form/field/StringField/index.tsx [hunk 2]
```diff
@@ -1,8 +1,8 @@
 import { ChangeEventHandler } from 'react'
-import { inputType } from '../../../utilities/types/formTypes'
+import { inputType } from '../../../../utilities/types/formTypes'
 import styles from './FormField.module.scss'
 
-type FormFieldProps = {
+type StringFieldProps = {
     field: inputType
     type: string
```
[the remark this answers is not in the document]
Comment (KevinLaCarrubba) [reply components/form/field/StringField/index.tsx]: left type in order to set type 'email' or 'text'.
set the type prop to check for either of these options only

File: components/form/field/StringField/index.tsx [hunk 3]
```diff
@@ -24,9 +24,9 @@ const FormField = ({
     required,
     title,
     minLength
-}: FormFieldProps) => {
+}: StringFieldProps) => {
     return (
-        <div className={styles.FormField}>
+        <div className={styles.StringField}>
             <input
                 type={type}
```
[the remark this answers is not in the document]
Comment (KevinLaCarrubba) [reply components/form/field/StringField/index.tsx]: call me a lumber jack the way i axe that shit

File: components/views/auth/Register.tsx [hunk 4]
```diff
@@ -64,7 +64,7 @@ const Register = () => {
                 <form onSubmit={handleSubmit}>
                     <fieldset>
                         <FormRow>
-                            <FormField
```
[the remark this answers is not in the document]
Comment (KevinLaCarrubba) [reply components/views/auth/Register.tsx:67]: kept type in string field but removed it from password field and added password field in on login page

**Review by arfusop — APPROVED** [review]

**Pull request merged**